src/components/Home/AptRequests.jsx

```diff
@@ -57,54 +56,71 @@ class AptRequests extends Component {
   }
 
   handleApprove(e) {
-    let slotId = null;
+    const time = e.currentTarget.getAttribute('time')
+    const email = e.currentTarget.getAttribute('email')
+    const instructorId = e.currentTarget.getAttribute('instructorId')
+    const slotId = e.currentTarget.getAttribute('slotId')
     let id = e.currentTarget.getAttribute('id');
     let requests = this.state.requests;
-    for (let i = 0; i < requests.length; i++) {
-      if (requests[i].id === id) {
-        axios.post(`/api/BookedApts`, {
-          "timeSlot": requests[i].time,
-          "studentName": requests[i].studentName,
-          "slotId": requests[i].slotId,
-          "visitorId": requests[i].visitorId,
-          "duration": 30
-        }).then((res) => {
-          axios.delete(`/api/Slots/${requests[i].slotId}`)
-          .then((r) => {
-          })
-          .catch((e) => {
-            console.log(e)
-          })
-        }).catch((err) => {
-          console.log(err)
-        })
-
 
+    let booked = requests.find((book) => {
+      return book.id === id
+    })
+    axios.post(`/api/BookedApts`, {
+      "timeSlot": booked.time,
+      "instructorId": booked.instructorId,
+      "studentName": booked.studentName,
+      "slotId": booked.slotId,
+      "visitorId": booked.visitorId,
+      "duration": 30
+    })
+    .then((res) => {
+      for (let i = 0; i < requests.length; i++) {
+        if (requests[i].time === time && requests[i].id !== booked.id) {
+          axios.post(`/api/AptRequests/replacedApt`, {
+            email: requests[i].email,
+            time: requests[i].time
+          })
+        }
       }
-    }
-    const email = e.currentTarget.getAttribute('email')
-    const time = e.currentTarget.getAttribute('time')
+      axios.delete(`/api/Slots/${booked.slotId}`)
+      .then((r) => {
+      })
+      .catch((e) => {
+        console.log(e)
+      })
+      console.log('instructorId', instructorId);
+    }).catch((err) => {
+      console.log(err)
+    })
+    
+    
+    
     axios.post(`/api/AptRequests/approveEmail`, {
```

Comment:
This email should go out inside the once the booked appointment has been created. Take this Axios call and put it within the `.then` of the above POST request